vars/copyContainer.groovy

```diff
@@ -10,34 +10,67 @@
  * Copies a container from one docker registry to another
  *
  * @param args A map of the following parameters
+ * @param args.recursiveCopy Copy all the tags of a sourceImage to destinationImage if 'true' and <IMAGE_TAG> is ignored in sourceImage/destinationImage, default to 'false'
  * @param args.sourceImage The Source Image name and tag <IMAGE_NAME>:<IMAGE_TAG> Eg: opensearch:1.3.2
  * @param args.sourceRegistry The source docker registry, currently supports 'DockerHub' or 'ECR'
  * @param args.destinationImage The Destination Image name and tag <IMAGE_NAME>:<IMAGE_TAG> Eg: opensearch:1.3.2
  * @param args.destinationRegistry The destination docker registry, currently supports 'DockerHub' or 'ECR'
  */
 void call(Map args = [:]) {
 
+    recursive_copy = args.recursiveCopy ?: false
+    source_image = args.sourceImage
+    source_image_no_tag = source_image.split(':')[0]
+    source_registry = args.sourceRegistry
+    destination_image = args.destinationImage
+    destination_image_no_tag = destination_image.split(':')[0]
+    destination_registry = args.destinationRegistry
 
     if (args.destinationRegistry == 'opensearchstaging' || args.destinationRegistry == 'opensearchproject') {
         def dockerJenkinsCredential = args.destinationRegistry == 'opensearchproject' ? "jenkins-production-dockerhub-credential" : "jenkins-staging-dockerhub-credential"
         withCredentials([usernamePassword(credentialsId: dockerJenkinsCredential, usernameVariable: 'DOCKER_USERNAME', passwordVariable: 'DOCKER_PASSWORD')]) {
             def dockerLogin = sh(returnStdout: true, script: "set +x && echo $DOCKER_PASSWORD | docker login --username $DOCKER_USERNAME --password-stdin").trim()
-            sh """gcrane cp ${args.sourceRegistry}/${args.sourceImage} ${args.destinationRegistry}/${args.destinationImage}; docker logout"""
+            gcraneCopy()
         }
     }
+
     if (args.destinationRegistry == 'public.ecr.aws/opensearchproject') {
         withCredentials([
             string(credentialsId: 'jenkins-artifact-promotion-role', variable: 'ARTIFACT_PROMOTION_ROLE_NAME'),
             string(credentialsId: 'jenkins-aws-production-account', variable: 'AWS_ACCOUNT_ARTIFACT')])
             {
                 withAWS(role: "${ARTIFACT_PROMOTION_ROLE_NAME}", roleAccount: "${AWS_ACCOUNT_ARTIFACT}", duration: 900, roleSessionName: 'jenkins-session') {
                     def ecrLogin = sh(returnStdout: true, script: "set +x && aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin ${args.destinationRegistry}").trim()
-                    sh """gcrane cp ${args.sourceRegistry}/${args.sourceImage} ${args.destinationRegistry}/${args.destinationImage}; docker logout ${args.destinationRegistry}"""
+                    gcraneCopy()
                 }
             }
     }
+
     if(args.destinationRegistry == 'public.ecr.aws/opensearchstaging') {
```

Comment:
Why is this different from `public.ecr.aws/opensearchproject` in terms of logging in?

Reply:
That is because the staging ecr and prod ecr are in 2 different accounts.
So different credentials.
Thanks.